libs/application/templates/document-provider-onboarding/src/lib/DocumentProviderOnboardingTemplate.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  ApplicationTemplate,
+  ApplicationTypes,
+  ApplicationContext,
+  ApplicationRole,
+  ApplicationStateSchema,
+} from '@island.is/application/core'
+import * as z from 'zod'
+
+type Events =
+  | { type: 'APPROVE' }
+  | { type: 'REJECT' }
+  | { type: 'SUBMIT' }
+  | { type: 'ABORT' }
+
+const nationalIdRegex = /([0-9]){6}-?([0-9]){4}/
+
+const contact = z.object({
+  name: z.string().nonempty(),
+  email: z.string().email().nonempty(),
+  phoneNumber: z.string().min(7),
+})
+
+//TODO: extend contact. Couldn't get it to work easily with contact.extend
+const applicant = z.object({
+  name: z.string().nonempty(),
+  email: z.string().email().nonempty(),
+  phoneNumber: z.string().min(7),
+  nationalId: z.string().refine((x) => (x ? nationalIdRegex.test(x) : false), {
+    message: 'Skrá þarf löglega kennitölu, með eða án bandstriks', //Question: how should error messages be translated?
+  }),
+  address: z.string().nonempty(),
+  zipCode: z.string().nonempty(),
+})
+
+const dataSchema = z.object({
+  applicant: applicant,
+  administrativeContact: contact,
+  technicalContact: contact,
+  rejectionReason: z.string(),
+  approvedByReviewer: z.enum(['APPROVE', 'REJECT']),
+})
+
+const DocumentProviderOnboardingTemplate: ApplicationTemplate<
+  ApplicationContext,
+  ApplicationStateSchema<Events>,
+  Events
+> = {
+  type: ApplicationTypes.DOCUMENT_PROVIDER_ONBOARDING,
+  name: 'Umsókn um að gerast skjalaveitandi',
+  dataProviders: [],
+  dataSchema,
+  stateMachineConfig: {
+    initial: 'draft',
+    states: {
+      draft: {
+        meta: {
+          name: 'Umsókn skjalaveitu',
+          progress: 0.25,
+          roles: [
+            {
+              id: 'applicant',
+              formLoader: () =>
+                import('../forms/DocumentProviderApplication').then((val) =>
+                  Promise.resolve(val.DocumentProviderOnboarding),
+                ),
+              actions: [
+                { event: 'SUBMIT', name: 'Staðfesta', type: 'primary' },
+              ],
+              write: 'all',
+            },
+          ],
+        },
+        on: {
+          SUBMIT: {
+            target: 'inReview',
+          },
+        },
+      },
+      inReview: {
+        meta: {
+          name: 'In Review',
+          progress: 0.5,
+          roles: [
+            {
+              id: 'reviewer',
+              formLoader: () =>
+                import('../forms/ReviewApplication').then((val) =>
+                  Promise.resolve(val.ReviewApplication),
+                ),
+              actions: [
+                { event: 'APPROVE', name: 'Samþykkja', type: 'primary' },
+                { event: 'REJECT', name: 'Hafna', type: 'reject' },
+              ],
+              read: 'all',
+              write: 'all',
```

Comment:
are you sure you want the reviewer to be able to review everything?

Reply:
ahh no... will change this to add only the message to the write privileges